compiler/hie-core/src/Development/IDE/Core/Compile.hs

```diff
@@ -138,25 +136,18 @@ addRelativeImport modu dflags = dflags
 
 mkTcModuleResult
     :: GhcMonad m
-    => InterfaceDirectory
-    -> TypecheckedModule
+    => TypecheckedModule
     -> m TcModuleResult
-mkTcModuleResult (InterfaceDirectory mbIfaceDir) tcm = do
-    session   <- getSession
-    (iface,_) <- liftIO $ mkIfaceTc session Nothing Sf_None details tcGblEnv
-    liftIO $ whenJust mbIfaceDir $ \ifaceDir -> do
-        let path = ifaceDir </> file tcm
-        createDirectoryIfMissing True (takeDirectory path)
-        writeIfaceFile (hsc_dflags session) (replaceExtension path ".hi") iface
-        -- For now, we write .hie files whenever we write .hi files which roughly corresponds to
-        -- when we are building a package. It should be easily decoupable if that turns out to be
-        -- useful.
-        hieFile <- runHsc session $ mkHieFile (tcModSummary tcm) tcGblEnv (fromJust $ renamedSource tcm)
-        writeHieFile (replaceExtension path ".hie") hieFile
+mkTcModuleResult tcm = do
+    session <- getSession
+    (iface, _) <- liftIO $ mkIfaceTc session Nothing Sf_None details tcGblEnv
+    hieFile <-
```

Comment:
This looks like a regression: You are now always producing hie files even if we do not use them.

Reply:
if producing the hie files is very costy, we can instead put the necessary arguments to produce them later into the TcModuleResult.

Reply:
Sounds good! You might need to be a bit careful that putting additional stuff in `TcModuleResult` does not make things more expensive since we force it to NF.